File: src/types/analysis.ts
export type SatelliteSource = 'IR' | 'WV' | 'VIS' | 'MW';

export interface TrackPoint {
  hours: number;
  latitude: number;
  longitude: number;
  wind_knots: number;
  uncertainty_km?: number;
}

export interface CycloneAnalysis {
  cyclone_detected: boolean;
  classification: { label: string; confidence: number };
  intensity: { wind_knots: number; pressure_hpa: number };
  center: { latitude: number; longitude: number };
  movement: { direction: string; speed_knots: number };
  observed_track?: TrackPoint[];
  track: TrackPoint[];
  explainability: { labels: string[]; heatmap_url?: string | null };
  model: { version: string; generated_at: string };
}

export interface AnalysisRequest {
  timestamp: string;
  latitude: number;
  longitude: number;
  sources: SatelliteSource[];
}

File: apps/api/app/schemas/analysis.py
from datetime import datetime
from pydantic import BaseModel, Field


class AnalysisRequest(BaseModel):
    timestamp: datetime
    latitude: float = Field(ge=-90, le=90)
    longitude: float = Field(ge=-180, le=180)
    sources: list[str] = Field(default_factory=list)


class TrackPoint(BaseModel):
    hours: int
    latitude: float
    longitude: float
    wind_knots: float
    uncertainty_km: float | None = None


class AnalysisResponse(BaseModel):
    cyclone_detected: bool
    classification: dict[str, float | str]
    intensity: dict[str, float]
    center: dict[str, float]
    movement: dict[str, float | str]
    observed_track: list[TrackPoint] = Field(default_factory=list)
    track: list[TrackPoint]
    explainability: dict[str, object]
    model: dict[str, str]
